community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/Transaction.java

```diff
@@ -19,7 +19,25 @@
  */
 package org.neo4j.internal.kernel.api;
 
-public interface Transaction extends Read, Write
+public interface Transaction
```

Comment:
Hmm, I don't know which existing interface you suggest that we use instead? `org.neo4j.graphdb.Transaction` is public, and `KernelTransaction` is on a lower level as far as I understand.

Reply:
It's completely fine and understandable that you need some kind of new interface for this kind of facade but just named it specifically what kind of Transaction it is to avoid confusions since it would be hard to read code around different kinds of transactions. CypherTransaction, etc would be better

Reply:
Problem is, it's just a transaction. Nothing cyphery about it. It's in essence a `KernelTransaction`, but that name is also taken. I cannot find a better name, if anyone else does, let's improve this later.